Tests/UnitTests/TrustEvaluatorTests.swift

```diff
@@ -0,0 +1,174 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+import XCTest
+@testable import ButtonMerchant
+
+class TrustEvaluatorTests: XCTestCase {
+    
+    var evaluator: TrustEvaluator!
+    
+    let system = TestSystem()
+    let device = TestDevice()
+    lazy var validTrust: SecTrust = {
+        var trust: SecTrust?
+        SecTrustCreateWithCertificates(TestCertificates.validCertChain as CFTypeRef, nil, &trust)
+        return trust!
+    }()
+    lazy var invalidTrust: SecTrust = {
+        var trust: SecTrust?
+        SecTrustCreateWithCertificates(TestCertificates.invalidCertChain as CFTypeRef, nil, &trust)
+        return trust!
+    }()
+    lazy var unpinnedTrust: SecTrust = {
+        var trust: SecTrust?
+        SecTrustCreateWithCertificates(TestCertificates.unpinnedCertChain as CFTypeRef, nil, &trust)
+        return trust!
+    }()
+    
+    override func setUp() {
+        system.device = device
+        evaluator = TrustEvaluator(system: system)
+    }
+    
+    func testInitialization_iOS10_certificateStrategy() {
+        // Arrange
+        device.testSystemVersion = "10.0"
+        
+        // Act
+        evaluator = TrustEvaluator(system: system)
+        
+        // Assert
+        XCTAssertNotNil(evaluator)
+        XCTAssertEqual(evaluator.strategy, ValidationStrategy.certificates)
+    }
+    
+    func testInitialization_iOS10_3_publicKeyStrategy() {
+        // Arrange
+        device.testSystemVersion = "10.3"
+        
+        // Act
+        evaluator = TrustEvaluator(system: system)
+        
+        // Assert
+        XCTAssertNotNil(evaluator)
+        XCTAssertEqual(evaluator.strategy, ValidationStrategy.publicKeys)
+    }
+    
+    func testInitialization_iOS12_publicKeyStrategy() {
+        // Arrange
+        device.testSystemVersion = "12.0"
+        
+        // Act
+        evaluator = TrustEvaluator(system: system)
+        
+        // Assert
+        XCTAssertNotNil(evaluator)
+        XCTAssertEqual(evaluator.strategy, ValidationStrategy.publicKeys)
+    }
+    
+    func testHandleChallenge_basicAuthMethod_performsDefaultHandling() {
+        // Arrange
+        let space = TestURLProtectionSpace()
+        space.authenticationMethod = NSURLAuthenticationMethodHTTPBasic
+        let challenge = TestURLAuthenticationChallenge(space)
+        
+        // Act
+        evaluator.handleChallenge(challenge) { disposition, _ in
+            
+            // Assert
+            XCTAssertEqual(disposition, URLSession.AuthChallengeDisposition.performDefaultHandling)
```

Comment:
It seems like this test could finish without executing this assertion. Looks like the completion is called sync, but if the implementation were to change (however unlikely) these tests could never fail.

Reply:
Done